modules/rostests/apitests/shell32/GetDisplayNameOf.cpp

```diff
@@ -0,0 +1,44 @@
+/*
+ * PROJECT:     ReactOS API tests
+ * LICENSE:     LGPL-2.1+ (https://spdx.org/licenses/LGPL-2.1+)
+ * PURPOSE:     Test for GetDisplayNameOf
+ * COPYRIGHT:   Copyright 2023 Mark Jansen <mark.jansen@reactos.org>
+ *              Copyright 2023 Doug Lyons <douglyons@douglyons.com>
+ */
+
+#include "shelltest.h"
+#include <stdio.h>
+#include <shellutils.h>
+
+START_TEST(GetDisplayNameOf)
+{
+    HRESULT hr;
+    CoInitializeEx(NULL, COINIT_APARTMENTTHREADED);
+
+    CComPtr<IShellFolder> spPanel;
+
+    hr = CoCreateInstance(CLSID_ControlPanel, NULL, CLSCTX_ALL,
+                          IID_PPV_ARG(IShellFolder, &spPanel));
+    ok_hr(hr, S_OK);
+    if (SUCCEEDED(hr))
+    {
+        STRRET ret, expected;
+
+        memset(&ret, 'a', sizeof(ret));
+        memset(&expected, 'a', sizeof(expected));
```

Comment:
I'm not sure this is a good idea to initialize a STRRET structure this way, but okay. (MSDN: https://learn.microsoft.com/en-us/windows/win32/api/shtypes/ns-shtypes-strret)